app/models/like.rb

```diff
@@ -1,4 +1,14 @@
 class Like < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :user, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  after_save :update_like_counter_post
+
+  def author=(user)
+    self.user_id = user.id
+  end
```

Comment:
- **Optional**: You have done well, I suggest you follow the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html). Precisely follow the `update_post_counter` example I suggested above, to make your `update_like_counter_post`.


 ```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```